src/woo_publications/publications/models.py

```diff
@@ -429,21 +431,25 @@ def revoke_own_documents(
             )
 
     def apply_retention_policy(self):
-        information_category = get_retention_informatie_category(
-            self.informatie_categorieen.all()
-        )
-        assert information_category is not None, (
-            "A publication must have at least one information category"
-        )
-        self.bron_bewaartermijn = information_category.bron_bewaartermijn
-        self.selectiecategorie = information_category.selectiecategorie
-        self.archiefnominatie = information_category.archiefnominatie
-        self.archiefactiedatum = localdate(
-            self.registratiedatum
-            + relativedelta(years=information_category.bewaartermijn)
-        )
-        self.toelichting_bewaartermijn = information_category.toelichting_bewaartermijn
-        self.save()
+        if self.publicatiestatus != PublicationStatusOptions.concept:
```

Comment:
I'd write this as a guard clause instead:

```suggestion
        if self.publicatiestatus == PublicationStatusOptions.concept:
            return
        ... # existing code
```